src/static/pages/retro/routes.js

```diff
@@ -26,10 +28,24 @@ router.get('/:room', authorization.restrictToLoggedIn, function(req, res, next)
 });
 
 // handles socket connections to /retro namespace
-function retroWebSocketHandler(client) {
-    console.log('New connection:', client.handshake.session);
-    client.emit('connection:accepted', 'Good to go');
+function retroWebSocketHandler(client, ns) {
+    client.on('join:retro', function(room) {
+        var userKey = client.handshake.session && client.handshake.session.userKey,
+            user = sessionManager.getUser(userKey);
 
+        if (!user || !retrosManager.hasAccess(user.username) || !retrosManager.isActiveRetro(room)) {
+            return;
+        }
+
+        setupSocket(client, room);
+    });
+
+    function setupSocket(clientSocket, room) {
+        clientSocket.join(room);
+
+        clientSocket.emit('connection:accepted', room);
+        ns.breadcast.to(room);
```

Comment:
ns.breadcast ?
Should it be clientSocket.broadcast.to(room).emit( someMessage)?